helper/resource/testing_new_import_state.go

```diff
@@ -109,12 +108,20 @@ func testStepNewImportState(ctx context.Context, t testing.T, helper *plugintest
 
 	logging.HelperResourceTrace(ctx, fmt.Sprintf("Using import identifier: %s", importId))
 
+	var priorIdentityValues map[string]any
+
 	// Append to previous step config unless using explicit inline Config, or ConfigFile, or ConfigDirectory
 	if testStepConfig == nil && step.ConfigFile == nil && step.ConfigDirectory == nil {
 		logging.HelperResourceTrace(ctx, "Using prior TestStep Config for import")
 		importConfig := cfgRaw
 
-		if kind.plannable() {
+		if kind.plannable() && kind.resourceIdentity() {
+			priorIdentityValues = identityValuesFromState(stateJSON, resourceName)
+			if len(priorIdentityValues) == 0 {
+				return fmt.Errorf("importing resource %s: expected prior state to have resource identity values, got none", resourceName)
+			}
```

Comment:
I believe this logic would need to be moved to another location outside the `if testStepConfig == nil && step.ConfigFile == nil && step.ConfigDirectory == nil` condition.

If you're providing a `step.ConfigFile`, it won't grab the prior identity values, then when we go to compare we'll end up comparing an empty map to whatever the identity actually is.

Reply:
Spot on. Fixed, with a new test.